### PR TITLE
Improve routes

### PR DESCRIPTION
**Pull request by yuniel-acosta**
-   Please make sure the below checklist is followed for Pull Requests.

-   [x ] [All continuous integration tests](https://github.com/jhipster/jhipster-vuejs/actions) are green
-   [x ] Tests are added where necessary
-   [x ] Documentation is added/updated where necessary
-   [x ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/jhipster-vuejs/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by atomfrede** [comment]
Great work @yuniel-acosta. One question/suggestion. Would it be possible to put the routes under the corresponding module? So not having all routes in one folder, but have the account routes in accounts etc.?

**Comment by yuniel-acosta** [comment]
@atomfrede It is possible, but I am not sure if it is a good option. I need the opinion of the community to reach an agreement and that everyone is satisfied

**Comment by yuniel-acosta** [comment]
@atomfrede, @pascalgrimaud, @nassimerrahoui 
Hello guys 
I need help with tests when using oauth2 authentication. I can't solve because it fails when importing routes

**Comment by pascalgrimaud** [comment]
@yuniel-acosta : did you manage to reproduce the issue locally ?
If you didn't try yet:
- generate a vuejs project with oauth2
- just launch npm test

If it passed locally, then, I'll try to find time tomorrow to have a look and reproduce. I can't right now, as I'm not with my computer

**Comment by yuniel-acosta** [comment]
@pascalgrimaud I can reproduce the errors locally and they are practically in the test files **account.service.spec.ts**, **jhi-navbar.component.spec.ts**, **error.component.spec.ts**

The error is when importing the module of the routes
`import router from '@/router';`

Error output:
 **TypeError: Cannot read property 'path' of undefined**
```
      17 | 
      18 | // prettier-ignore
    > 19 | export default new Router({
            |                ^
      20 |   mode: 'history',
      21 |   routes: [
      22 |     {
```

File: generators/client/templates/vue/src/main/webapp/app/router/admin.ts.ejs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      name: '<%= jhiPrefixCapitalized %>UserEdit',
+      component: <%= jhiPrefixCapitalized %>UserManagementEditComponent,
+      meta: { authorities: [Authority.ADMIN] }
+    },
+    {
+      path: '/admin/user-management/:userId/view',
+      name: '<%= jhiPrefixCapitalized %>UserView',
+      component: <%= jhiPrefixCapitalized %>UserManagementViewComponent,
+      meta: { authorities: [Authority.ADMIN] }
+    }<% } %>,
```
Comment (pascalgrimaud) [inline generators/client/templates/vue/src/main/webapp/app/router/admin.ts.ejs:48]:
```suggestion
    },<% } %>
```

Comment (pascalgrimaud) [reply generators/client/templates/vue/src/main/webapp/app/router/admin.ts.ejs:48]: With OAuth2 application, the `admin.ts` file is not correct:

```typescript
export default [
  ,
  {
    path: '/admin/docs',
    name: 'JhiDocsComponent',
    component: JhiDocsComponent,
    meta: { authorities: [Authority.ADMIN] },
  },
```

**Comment by pascalgrimaud** [comment]
> ok, @pascalgrimaud All work is done, @atomfrede suggests that I put the route files inside the corresponding module, personally I don't really like the idea but I need to know your opinion to make the relevant changes.

I don't like too. I find this new structure is good enough. So I'd prefer to keep like this, inside `router/**`
So I think we are ready to merge this.

Are you OK with this @atomfrede ?
I'd like to merge this and make a new release. So I can go on my work on merging this blueprint into main generator-jhipster :-)

**Comment by atomfrede** [comment]
Sure. Let's merge it

**Comment by pascalgrimaud** [comment]
For your information, I already have a Vue.js project, generated with generator-jhipster. It works on my machine. But some options don't work yet.
I didn't finish all the tests, the CI, and the sub module `page`

So I need to include your change here in my branch !

**Comment by pascalgrimaud** [comment]
thanks a lot @yuniel-acosta 
FYI, the new release has been done :)

**Comment by yuniel-acosta** [comment]
You are welcome.
It has been a pleasure for me to contribute to this incredible project.
I hope to contribute more in the future

**Comment by atomfrede** [comment]
Thanks for the work @yuniel-acosta!

**Pull request merged**